vendor/github.com/containers/image/pkg/authentication/authentication.go

```diff
@@ -0,0 +1,254 @@
+package authentication
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/errors"
+)
+
+type encodedAuth struct {
+	Auth string `json:"auth"`
+}
+
+type allAuths struct {
+	Auths map[string]encodedAuth `json:"auths"`
+}
+
+// SetAuthentication stores the base64 encoding of username and password
+// in the auth file if the credentials are authenticated
+func SetAuthentication(username, password, server string) error {
+	path, err := getPathToAuth()
+	if err != nil {
+		return errors.Wrapf(err, "error getting path to auth file")
+	}
+
+	server = getFullURL(server)
+
+	isValid, err := checkAuthentication(username, password, server)
+	if err != nil {
+		return errors.Wrapf(err, "An error occured while authenticating credentials")
```

Comment:
I think we standardized on lower case error messages.

Reply:
fixed